src/js/components/NetworkErrorNotice.tsx

```diff
@@ -15,21 +16,39 @@ const MAX_BACKOFF = 128
 
 export default function NetworkErrorNotice({error}: Props) {
   const dispatch = useDispatch()
-  const spaceId = useSelector(Current.getSpaceId)
+  const connId = useSelector(Current.getConnectionId)
+  const status = useSelector(ConnectionStatuses.get(connId))
   const [count, setCount] = useState(0)
+  const statusRef = useRef(status)
+  const connRef = useRef(connId)
 
-  const retry = () => dispatch(initSpace(spaceId))
-  const dismiss = () => dispatch(Notice.dismiss())
+  const retry = () => {
+    setCount(0)
+    dispatch(checkStatus())
+  }
+  const dismiss = () => {
+    dispatch(Notice.dismiss())
+  }
 
   useEffect(() => {
+    if (connId === connRef.current) {
+      statusRef.current = status
+      if (status === "disconnected") dismiss()
+    }
+  }, [status, connId])
```

Comment:
this effect does two things. it updates the statusRef used when the component exits, and it will also force the component to exit if it sees that status has changed to "disconnected". It will only do those things if the connection matches the one set when the component was first created (and is referencing throughout it's lifecycle).

I had originally not paid any attention to the connId for the status monitoring, and ended up in situations where we if we switch the connection then this component would begin to operate against the wrong conn.